internal/controller/pkg/revision/reconciler.go

```diff
@@ -549,40 +556,45 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 		}
 	}
 	if err != nil {
-		pr.SetConditions(v1.Unhealthy())
-		_ = r.client.Status().Update(ctx, pr)
 		log.Debug(errParsePackage, "error", err)
 
 		err = errors.Wrap(err, errParsePackage)
+		pr.SetConditions(v1.Unhealthy().WithMessage(err.Error()))
+		_ = r.client.Status().Update(ctx, pr)
+
 		r.record.Event(pr, event.Warning(reasonParse, err))
 		return reconcile.Result{}, err
 	}
 
 	// Lint package using package-specific linter.
 	if err := r.linter.Lint(pkg); err != nil {
-		pr.SetConditions(v1.Unhealthy())
+		log.Debug(errLintPackage, "error", err)
+
+		err = errors.Wrap(err, errLintPackage)
+		pr.SetConditions(v1.Unhealthy().WithMessage(err.Error()))
 		_ = r.client.Status().Update(ctx, pr)
 
+		r.record.Event(pr, event.Warning(reasonLint, err))
```

Comment:
Do you think if we have too much of similar lines that could be refactored into a dedicated function?

Reply:
They are not really equal but differ in details. Also this is preexisting in the code. I have only moved a few lines consistently to make sense.